velox/functions/sparksql/RegisterArithmetic.cpp

```diff
@@ -81,6 +81,13 @@ void registerArithmeticFunctions(const std::string& prefix) {
   registerFunction<HypotFunction, double, double, double>({prefix + "hypot"});
   registerFunction<sparksql::Log2Function, double, double>({prefix + "log2"});
   registerFunction<sparksql::Log10Function, double, double>({prefix + "log10"});
+  registerFunction<
+      WidthBucketFunction,
```

Comment:
width_bucket function supports the same data type, not only double

Reply:
thanks , i updated to register for more type - double, int64 and int32, what other type do you think still needed ?

Reply:
> width_bucket function supports the same data type, not only double

@jinchengchenghh Is that the case? I find for the first to third argument, only (DoubleType, YearMonthIntervalType, DayTimeIntervalType) are possible. Perhaps it's sufficient to support double.

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L1844-L1848

Reply:
agree with @rui-mo looks double is enough @jinchengchenghh , we can remove right now for `int64_t` and `int32_t`

Reply:
ok thanks!